### PR TITLE
FEAT-#96: Add python-versioneer to get exact unidist version

### LINKED ISSUE #96: Set conf value "version" to a concrete value
When building docs we have `WARNING: conf value "version" should not be empty for EPUB3` that should be fixed.

https://github.com/python-versioneer/python-versioneer should be used.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #96  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by YarShev** [comment]
Blocked by #106.

**Comment by YarShev** [comment]
@no-ponomarev, @prutskov, ping to review

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+unidist/_version.py export-subst
```
Comment (prutskov) [inline .gitattributes:1]: Should we also add `* text=auto` as Modin does?

File: docs/conf.py [hunk 2]
```diff
@@ -8,18 +8,19 @@
...
 
-# The full version, including alpha/beta/rc tags
-release = "0.0.1"
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+import unidist  # noqa: E402
 
+# The short project version like 1.1
+version = f"{unidist.__version__}"
+# The full project version like 1.1.0rc1
+release = version
```
Comment (prutskov) [inline docs/conf.py:23]: I didn't quite get it. Why version and release are equal?

File: setup.cfg [hunk 3]
```diff
@@ -2,3 +2,14 @@
 max-line-length = 88
 ignore = E203, E266, E501, W503
 select = B,C,E,F,W,T4,B9
+per-file-ignores =
+    versioneer.py:T001
+    unidist/_version.py:T001
```
Comment (prutskov) [inline setup.cfg:7]: We can just skip autogenerated files, there is no point to check them:
```suggestion
exclude =
    versioneer.py,
    unidist/_version.py
```

File: setup.cfg [hunk 4]
```diff
@@ -2,3 +2,14 @@
...
+
+[versioneer]
+VCS = git
+style = pep440
+versionfile_source = unidist/_version.py
+versionfile_build = unidist/_version.py
+tag_prefix =
```
Comment (prutskov) [inline setup.cfg:14]: Does this parameter required?

File: .gitattributes (hunk 1 above)
Comment (YarShev) [reply .gitattributes:1]: Yes, we should.

File: docs/conf.py (hunk 2 above)
Comment (YarShev) [reply docs/conf.py:23]: This is for sphinx settings. Each var is responsible for different concepts.

File: setup.cfg (hunk 4 above)
Comment (YarShev) [reply setup.cfg:14]: Yes

File: setup.cfg (hunk 3 above)
Comment (YarShev) [reply setup.cfg:7]: I think we can keep as it is. There are no many different warnings. If we want to keep the whole code base formatted, we likely want to check these too.

File: .gitattributes (hunk 1 above)
Comment (YarShev) [reply .gitattributes:1]: added

**Review by prutskov — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**